### PR TITLE
meson: Address various issues with the meson build system

### PR DESCRIPTION
**Pull request by dgsga**
This changeset refines the meson build system:

- Enables quota support on all flavours of linux and BSD (including macOS)
- Adds the quota provider to the configuration summary
- Adds a user option to disable LDAP support
- Sets dependencies according to user configuration
- Improves the syntax of the ACL macro

### CODE REVIEW

**Comment by dgsga** [comment]
@rdmark, I will backport this PR to 2.4 on approval.

File: etc/afpd/unix.h [hunk 1]
```diff
@@ -202,4 +210,8 @@ extern int setfilunixmode   (const struct vol *, struct path*, const mode_t);
 extern int setfilowner      (const struct vol *, const uid_t, const gid_t, struct path*);
 extern void accessmode      (const AFPObj *obj, const struct vol *, char *, struct maccess *, struct dir *, struct stat *);
 
+#ifdef AFS	
+    #define accessmode afsmode
+#endif
```
Comment (rdmark) [inline etc/afpd/unix.h:215]: I thought we got rid of all traces of Andrew File System... but I assume this is something else?

File: etc/cnid_dbd/meson.build [hunk 2]
```diff
@@ -1,73 +1,81 @@
...
-    'dbd_lookup.c',
-    'dbd_rebuild_add.c',
-    'dbd_resolve.c',
-    'dbd_search.c',
-    'dbd_update.c',
-    'dbif.c',
-    'main.c',
-    'pack.c',
-]
+if use_dbd_backend
```
Comment (rdmark) [inline etc/cnid_dbd/meson.build:1]: Is there an argument to be made to always enable the dbd backend?

The argument to always enable: It is the one single stable and full-featured CNID backend presently.

The argument to make it optional: The potential of creating a new CNID backend in the future (e.g. sqlite) which will become the default.

**Review by rdmark — APPROVED** [review]
Huge improvement!

**Pull request merged**